embeddings.mdx

```diff
@@ -0,0 +1,147 @@
+---
+title: Embeddings
+description: Automatically calculate embeddings from existing features
+comingSoon: true
+---
+
+import { TipInfo } from '@/components/Tip'
+
+---
+
+Embedding models are generally used to calculate a [vector feature](/docs/feature-types#vectors). Chalk includes
+built-in support for common embedding models, or can define your own embedding model through a resolver.
+
+## Built-In Embedding Functions
+
+Chalk includes built-in support for both open-source and hosted embedding models via the `embedding` function.
+We recommend using this function when possible, as Chalk will automatically handle batching and retries,
+and you don't need to specify the vector size. The main arguments for this function are:
+
+1.  `feature` (required): The name of feature that is used to generate the embedding. This feature must have a
+    string type. If you would like to use multiple features as input, you can define a [resolver](/docs/resolver) to map
+    these features into one that will be used as the input.
+1.  `provider` (required): The embedding model provider. The currently supported providers are `sentence-transformers`, `instructor`,
+    `openai`, or `cohere`. Chalk may add more providers in the future.
+1.  `model` (optional): The name of the model to use. Each model family has a default model, which will be used if this
+    argument is not specified.
+1.  `ttl` (optional): The duration for which the embedding will be cache. By default, the embedding vector will be
+    cached for the same TTL as the `feature`. If you would like different behavior,
+    you can specify this argument explicitly.
+
+For the complete signature, please see the [api docs](/api-docs#embedding).
+
+
+
+### Sentence Transformers
+
+Chalk supports all models that are part of the [sentence-transformers](https://www.sbert.net/) framework.
+The default model for this provider is `all-MiniLM-L6-v`. The full list of models is available
+[here](https://www.sbert.net/docs/pretrained_models.html).
+
+```python
+from chalk.features import embedding, features, Vector
+
+@features
+class Document:
+    content: str
+    embedding: Vector = embedding(
+        feature="content",
+        provider="sentence-transformers",
+        model="all-MiniLM-L6-v",  # optional, defaults to all-MiniLM-L6-v
+    )
```

Comment:
I think this could be nicer as 

```suggestion
    embedding: Vector = embedding(
        feature=_.content,
        provider="sentence-transformers",
        model="all-MiniLM-L6-v",  # optional, defaults to all-MiniLM-L6-v
    )
```

or

```suggestion
    embedding: AllMiniLML6V[_.content]
```

but certainly up to you on how you want to present it

Reply:
Hmm, personally I'm still warming up to the underscore syntax (partially because it is not type checked; partially because it feels too magical haha). Happy to revisit this once we have docs out on how to use underscore.

Reply:
even without the underscore, this could be written as `embedding: AllMiniLML6V["content"]`, matching the string version in the docs right now